### ISSUE TITLE
[Help] Missing Data on one Account - Other Works

### ISSUE THREAD

**Issue by tobru** [offen/offen#838]
I'm running one Offen instance, serving multiple domains:

- https://offen.example1.com
- https://offen.example2.com

Each domain has its own Offen account.

The script is included in the header of example1.com and example2.com, each with the unique account ID.

On example1.com, I see analytics data arriving, but not on example2.com. Looking at the browser console, I can see that Offen is properly loaded, and the consent has been given.

![Screenshot_20240518_230234](https://github.com/offen/offen/assets/1695957/8c28197b-1481-4d25-a076-60ea39c7f9a3)

![Screenshot_20240518_230339](https://github.com/offen/offen/assets/1695957/b1e7cec0-ff69-442e-aa55-58cbb2e73c02)

I'm not sure how I can figure out if and what I'm potentially doing wrong. Any ideas how I can proceed to find out what's wrong?

**Comment by m90** [comment]
This is very hard for me to tell. Are there errors in your console or Auditorium when accessing the account that doesn't display data?

**Comment by tobru** [comment]
> This is very hard for me to tell.

I'm sorry I couldn't offer more information right now, I'm a bit lost where to look exactly.

> Are there errors in your console or Auditorium when accessing the account that doesn't display data?

I just discovered that when opening the Auditorium on the Domain/Account which doesn't work I get:

```
GET https://offen.example2.com/api/events

{"error":"user cookie: received no or blank identifier","status":400}
```

**Comment by tobru** [comment]
I took a look into the database, the `events` table seems to fill up, but nothing is displayed in the Auditorium.

**Comment by m90** [comment]
> GET https://offen.example2.com/api/events

That's pretty odd as this call should only be made from within an user facing auditorium, the operator facing one should call `/accounts/<accountId>`. What's the URL you are looking at when receiving no data, and what's the one when you are receiving data?

**Comment by tobru** [comment]
On these URLs it works and I can see data:

* `https://offen.domain1.ch/auditorium/`
* `https://offen.domain1.ch/auditorium/<account-id-1>/`

And on these URLs no data is shown:

* `https://offen.domain2.com/auditorium/`
* `https://offen.domain2.com/auditorium/<account-id-2>/`

Both are pointing to the same Offen instance, just with a different DNS name.

**Comment by m90** [comment]
So this issue is about the user facing view (up until now I was under the impression it's the operator facing one). Does data show up when you log in as the operator and access the account?

**Comment by tobru** [comment]
Ooops, GitHub has eaten the URLs, I updated the comment. There is no data shown, neither in the public view, nor in the operator view for domain2. For domain1, I see data in both views.

**Comment by m90** [comment]
Looking at what's going on I can see data is being sent, persisted and also synced back to the Auditorium but fails to display.

This could be two things:

- your "other" account sends unexpected data that fails to decrypt for reasons I do not know yet
- for whatever reason, your account has malformed key data, which means data fails to decrypt

I've never seen this and it's pretty mysterious, but I'll see if I can debug this further in the next few days.

**Comment by m90** [comment]
So what's causing this is that on domain2 you are using a canonical link of `/` when domain1 is using `https://domain1.com/`. Further down the line, this creates an event payload that is considered invalid and will thus be skipped.

While [partial hrefs are usually discouraged](https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls?hl=en&visit_id=638518323754200859-377075974&rd=1#rel-canonical-link-method) this should still be supported, so I'll see when I can get a fix up.

In the meantime, you can work around this by either 1. removing the canonical link altogether or 2. use a full link just like you do in domain1

**Comment by tobru** [comment]
Thank you very much for investigating and resolving the issue. That's greatly appreciated :rocket:

**Comment by m90** [comment]
This is fixed in [v1.4.2](https://github.com/offen/offen/releases/tag/v1.4.2)
